packages/ra-data-simple-rest/src/index.ts

```diff
@@ -10,7 +10,78 @@ import {
     UPDATE_MANY,
     DELETE,
     DELETE_MANY,
-} from 'react-admin';
+} from 'ra-core';
+
+interface Pagenation {
+    page: number;
+    perPage: number;
+}
+
+interface Sort {
+    field: string;
+    order: 'ASC' | 'DESC';
+}
+
+interface GetListRequestParams {
+    pagination: Pagenation;
+    sort: Sort;
+    filter: {
+        [key: string]: any;
+    };
+}
+
+interface IdParams {
+    id: number | string;
+}
+interface IdsParams {
+    ids: (number | string)[];
+}
+
+interface GetOneRequestParams extends IdParams {}
+
+interface GetManyRequestParams extends IdsParams {}
+
+interface GetManyReferenceRequestParams extends IdParams {
+    pagination: Pagenation;
+    sort: Sort;
+    filter: {
+        [key: string]: any;
+    };
+    target: string;
+}
+interface UpdateRequestParams extends IdParams {
+    data: any;
+}
+interface UpdateManyRequestParams extends IdsParams {
+    data: any;
+}
+interface CreateRequestParams {
+    data: any;
+}
+interface DeleteRequestParams extends IdParams {}
+interface DeleteManyRequestParams extends IdsParams {}
+type RequestParams =
```

Comment:
users can send their own request types, so the list should not be limited.

Reply:
ra-data-simple-rest appears to accept no other params type.
What about changing name "RequestParams" to "DataSimpleRestRequestParams"?

Reply:
That's right, ra-data-simple-rest accepts no other param, but the type that react-admin must use should allow additional request types. I suppose that means adding a `default` case in ra-data-simple-rest to throw an error in case of unsupported type.